### PR TITLE
Restrict elements that can be tracked by amp-analytics tag inside a frame

### PR DESCRIPTION
**Pull request by avimehta**
Related to #5657

### CODE REVIEW

**Comment by avimehta** [comment]
I still need to figure out how the parent element of the iframe is going to be tracked. One option is to search by tag name for scope=closest.

**Comment by dvoytenko** [comment]
It's fine to move parent issues to a separate PR. I still prefer `:root` or `:host` type of selector though.

File: extensions/amp-analytics/0.1/instrumentation.js [hunk 1]
```diff
@@ -281,14 +279,16 @@ export class InstrumentationService {
...
    */
-  ensureClickListener_() {
-    if (!this.clickHandlerRegistered_) {
-      this.clickHandlerRegistered_ = true;
-      this.win_.document.documentElement.addEventListener(
-          'click', this.onClick_.bind(this));
+  ensureClickListener_(analyticsElement) {
+    const doc = analyticsElement.ownerDocument;
+    if (!doc[CLICK_LISTENER_REGISTERED_]) {
+      doc[CLICK_LISTENER_REGISTERED_] = true;
```
Comment (mkhatib) [inline extensions/amp-analytics/0.1/instrumentation.js]: @dvoytenko, @avimehta I am rebasing the other analytics-doc PR #5693, is this still needed in an ampdoc world?

Comment (avimehta) [inline extensions/amp-analytics/0.1/instrumentation.js]: Depends. Do the clicks from shadow dom get surfaced to the parent document? If not, we need to listen for clicks on shadowdom's doc as well.

Comment (mkhatib) [inline extensions/amp-analytics/0.1/instrumentation.js]: Clicks do [bubble as far as I can tell, with few quirks](https://developers.google.com/web/fundamentals/getting-started/primers/shadowdom#events) (if target is a shadow dom it is hidden). [Demo jsbin](http://jsbin.com/wafenu/edit?html,console,output).

Comment (dvoytenko) [inline extensions/amp-analytics/0.1/instrumentation.js]: @mkhatib I think your implementation is correct. But please do make sure with @avimehta we handling it right - i.e. please identify a test case to confirm against and do confirm.

**Pull request merged**